### PR TITLE
Populate resource to OTLP proto data

### LINKED ISSUE #757: Resource is not populated to OTLP proto data
In previous PR https://github.com/open-telemetry/opentelemetry-cpp/pull/706, resource is passed to exporter recordable structure, but it is still not populated to `proto::collector::trace::v1::ExportTraceServiceRequest`, so the OTLP receiver will not get any resource from the OpenTelemetry C++ SDK.

**Steps to reproduce**
Run example_otlp and OpenTelemetry collector with OTLP collector.

**What is the expected behavior?**
We should see "Resource labels" in the receive data.

**What is the actual behavior?**
No "Resource labels" sent to OpenTelemetry collector.

**Additional context**

### PR DESCRIPTION
**Pull request by ThomsonTan**
Fixes #757 

## Changes

Populate resource to OTLP proto data and also update the readme for OTLP receiver with easier instructions to run and test it end-to-end with docker.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: exporters/otlp/src/otlp_exporter.cc [hunk 1]
```diff
@@ -28,8 +28,18 @@ void PopulateRequest(const nostd::span<std::unique_ptr<sdk::trace::Recordable>>
...
     *instrumentation_lib->add_spans() = std::move(rec->span());
+
+    if (!has_resource)
+    {
+      std::unique_ptr<proto::resource::v1::Resource> proto_resource{
+          new proto::resource::v1::Resource};
+
+      rec->PopulateProtoResource(proto_resource.get());
+
+      resource_span->set_allocated_resource(proto_resource.release());
```
Comment (lalitb) [inline exporters/otlp/src/otlp_exporter.cc:40]: nit - Just wondering if line 35 - 40 can be replaced with single line:

``
resource_span->mutable_resource() = rec->ProtoResource();
``

using older implementation of `ProtoResource`.

**Review by lalitb — APPROVED** [review]
Thanks for adding the support. Looks good, perhaps we can add unit tests for the resources too.

File: exporters/otlp/src/otlp_exporter.cc (hunk 1 above)
Comment (ThomsonTan) [reply exporters/otlp/src/otlp_exporter.cc]: I chose `set_allocated_resource` which could avoid allocate the `Resource` twice or more, but yes, it look simpler with `mutable_resource`. Updated the PR to the older implementation of `ProtoResource` as this optimization may not deserve the extra complexity.

**Comment by ThomsonTan** [comment]
> Thanks for adding the support. Looks good, perhaps we can add unit tests for the resources too.

Added unit resource test to OTLP recordable test case. We may need more tests if we'd like to validate the OTLP exported result.

**Pull request merged**